### PR TITLE
Add running test-log-collector to bash trap

### PR DESCRIPTION
**Pull request by aerfio**
**Description**

Changes proposed in this pull request:

- fix badly written code which prevented test-log-collector from actually working when some kind of test failed

### CODE REVIEW

File: vpath/pjtester.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+pjNames:
+  - pjName: "pre-master-kyma-gke-integration"
+  - pjName: "post-master-kyma-gke-integration"
```
Comment (aerfio) [inline vpath/pjtester.yaml:3]:
```suggestion
pjNames:
  - pjName: pre-master-kyma-gke-integration
  - pjName: post-master-kyma-gke-integration
```

File: prow/scripts/cluster-integration/kyma-gke-integration.sh [hunk 2]
```diff
@@ -296,25 +314,13 @@ if [ -n "$(kubectl get  service -n kyma-system apiserver-proxy-ssl --ignore-not-
     IP_ADDRESS=${APISERVER_IP_ADDRESS} DNS_FULL_NAME=${APISERVER_DNS_FULL_NAME} "${TEST_INFRA_CLUSTER_INTEGRATION_SCRIPTS}/create-dns-record.sh"
 fi
 
+enableTestLogCollector=true
```
Comment (dekiel) [inline prow/scripts/cluster-integration/kyma-gke-integration.sh:317]: Add some short comment why collector is enabled in this place.

File: prow/scripts/cluster-integration/kyma-gke-integration.sh [hunk 3]
```diff
@@ -123,10 +125,26 @@ cleanup() {
...
-trap cleanup EXIT INT
+runTestLogCollector(){
+    if [ "${enableTestLogCollector}" = true ] ; then
+        if [[ "$BUILD_TYPE" == "master" ]]; then
+            shout "Install test-log-collector"
+            date
+            export PROW_JOB_TYPE="post-master-kyma-gke-integration"
+            ( 
+                # shellcheck disable=SC2030
+                export TEST_INFRA_SOURCES_DIR LOG_COLLECTOR_SLACK_TOKEN PROW_JOB_TYPE
```
Comment (dekiel) [inline prow/scripts/cluster-integration/kyma-gke-integration.sh:136]: You don't need to export variables again in subshell. Once exported they will be copied to all child processes.

File: development/test-log-collector/cmd/app/mainerr.go [hunk 4]
```diff
@@ -27,15 +27,14 @@ import (
...
-	"github.com/kyma-project/test-infra/development/test-log-collector/pkg/hyperscaler"
-
 	"github.com/kyma-project/test-infra/development/test-log-collector/pkg/resources/clustertestsuite"
 	octopusTypes "github.com/kyma-project/test-infra/development/test-log-collector/pkg/resources/clustertestsuite/types"
 )
 
 type config struct {
 	SlackToken     string
 	ConfigLocation string
+	ProwJobType    string `envconfig:"default=not-specified"` // post-master-kyma-gke-integration etc
```
Comment (dekiel) [inline development/test-log-collector/cmd/app/mainerr.go:37]: Variable name should be rather ProwJobName. ProwJob types are presubmit, postsubmit or periodic.

File: development/test-log-collector/chart/test-log-collector/templates/job.yaml [hunk 5]
```diff
@@ -23,7 +23,9 @@ spec:
             - name: APP_CONFIG_LOCATION
               value: /config/config.yaml
             - name: APP_SLACK_TOKEN
-              value: {{ .Values.slackToken }}  
+              value: {{ .Values.slackToken }}
```
Comment (dekiel) [inline development/test-log-collector/chart/test-log-collector/templates/job.yaml:26]: Please make sure value is not printed in any logs.

**Review by dekiel — APPROVED** [review]

**Review by kazydek — APPROVED** [review]
🎉

**Pull request merged**